pages/admin/request/create-new.tsx

```diff
@@ -302,7 +301,7 @@ export default function CreateNew() {
       return;
     }
 
-    if (!paymentDetails.paymentMethod) {
+    if (!values.paymentInformation.paymentMethod) {
```

Comment:
I think we should be able to remove this check and instead just use the validation schema to validate `values` which will ensure its non-null. For example, I do this [here](https://github.com/uwblueprint/richmond-centre-for-disability/blob/ad10ae7e3330a8c54dd60b163b9d80a52e4b3bf5/pages/admin/request/create-new.tsx#L272). Once we merge eveything together we won't have to implement this twice but depending on how long it takes for the other prs to get merged it might be a good idea to implement it here now anyways to get rid of the explicit checks. I believe the same changes can be made in the other files too.

Reply:
Curious why we do
https://github.com/uwblueprint/richmond-centre-for-disability/blob/ad10ae7e3330a8c54dd60b163b9d80a52e4b3bf5/pages/admin/request/create-new.tsx#L272 in this file?

Wouldn't any value that reaches this point be already validated since the submit button is disabled if the frontend validation fails? And if we wanted to add backend validation I think it would go in the resolver instead.

Reply:
Yeah, at this point the validation should never fail. The reason we still call it here is because `.validate()` returns an object with all of the validation constraints enforced in the return type. This means if in our validation schema we require a field to be non-null, that field in the result of `.validate()` will have a non-null type. Without doing this, we get a bunch of linting errors since the original `values` is a type which allows fields to be null, because this is the same type as our initial values in formik.